embedchain/embedchain.py

```diff
@@ -294,9 +302,16 @@ def dry_run(self, input_query):
         """
         if config is None:
             config = QueryConfig()
-        context = self.retrieve_from_database(input_query)
-        prompt = self.generate_prompt(input_query, context)
+        contexts = self.retrieve_from_database(input_query, config)
+        prompt = self.generate_prompt(input_query, contexts)
         return prompt
+    
```

Comment:
I think here, two functionalities are getting merged.

@cachho : can you keep this PR for number of docs in context, and create a new PR for reset or vice versa.

But the idea is to have different PR for both.